packages/@glimmer/syntax/test/template-locals-test.ts

```diff
@@ -35,6 +35,13 @@ QUnit.test('it works', function (assert) {
 
     <this.dynamicAngleComponent>
     </this.dynamicAngleComponent>
+
+    <ComponentYieldingContextual as |hash|>
+      <hash.some as |some|>
+        <some.other/>
```

Comment:
While we're here, let's add another couple that *should* be:

```suggestion
      <InsideHash />
      <hash.some as |some|>
        {{inside-another-hash}}
        <some.other/>
```

Then we can include these in the locals as well. Basically: I want to make sure that our block-yielding behavior handles *both* the yielded content and the regular content (and that it doesn't regret).